Neos.ContentRepository.Core/Classes/Service/ContentStreamPruner.php

```diff
@@ -129,13 +131,29 @@ private function findUnusedAndRemovedContentStreams(): ContentStreams
         }
 
         // Step 3: Check for removed content streams which we do not need anymore transitively
+        $allContentStreamEventStreamNames = $this->findAllContentStreamEventNames();
+
         $removedContentStreams = [];
-        foreach ($allContentStreams as $contentStream) {
-            if ($contentStream->removed && !array_key_exists($contentStream->id->value, $transitiveUsedStreams)) {
-                $removedContentStreams[] = $contentStream;
+        foreach ($allContentStreamEventStreamNames as $streamName) {
+            $removedContentStream = substr($streamName->value, strlen(ContentStreamEventStreamName::EVENT_STREAM_NAME_PREFIX));
+            if (!array_key_exists($removedContentStream, $transitiveUsedStreams)) {
+                $removedContentStreams[] = $streamName;
             }
         }
 
-        return ContentStreams::fromArray($removedContentStreams);
+        return $removedContentStreams;
+    }
+
+    /**
+     * @return list<StreamName>
+     */
+    private function findAllContentStreamEventNames(): array
+    {
+        $events = $this->eventStore->load(VirtualStreamName::forCategory(ContentStreamEventStreamName::EVENT_STREAM_NAME_PREFIX));
+        $allContentStreamEventStreamNames = [];
+        foreach ($events as $eventEnvelope) {
+            $allContentStreamEventStreamNames[$eventEnvelope->streamName->value] = true;
+        }
+        return array_map(StreamName::fromString(...), array_keys($allContentStreamEventStreamNames));
```

Comment:
more of a nit pick, but "ContentStreamEventNames" is not correct.
What about doing the mapping to the `ContentStreamId` here like so:

```suggestion
    /**
     * @return list<ContentStreamId>
     */
    private function findAllContentStreamIds(): array
    {
        $events = $this->eventStore->load(VirtualStreamName::forCategory(ContentStreamEventStreamName::EVENT_STREAM_NAME_PREFIX));
        $allContentStreamEventStreamNames = [];
        foreach ($events as $eventEnvelope) {
            $allContentStreamEventStreamNames[$eventEnvelope->streamName->value] = true;
        }
        return array_map(static fn (string $streamName) => substr($streamName, strlen(ContentStreamEventStreamName::EVENT_STREAM_NAME_PREFIX)), array_keys($allContentStreamEventStreamNames));
```
?

I know, this means that we do a couple of string operations for content streams that are not relevant above, but I don't think that this is measurable

Reply:
the thing is we convert it later to `eventStream` things again ... so i rather keep the return value this way... maybe we can find a better name for `findAllContentStreamEventNames`?